py/server/deephaven/updateby.py

```diff
@@ -0,0 +1,280 @@
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / time_scale_ticks)
+        ema_next = a * ema_last + (1 - a) * value
+
+    Args:
+        time_scale_ticks (int): the decay rate in ticks
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col = col"
+        op_control (OperationControl): defines how special cases should behave, when None, the default OperationControl
+            settings as specified in :meth:`~OperationControl.__init__` will be used
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        if op_control is None:
+            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.Ema(time_scale_ticks, *cols))
+        else:
+            return UpdateByOperation(
+                j_updateby_op=_JUpdateByOperation.Ema(op_control.j_op_control, time_scale_ticks, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a tick-decay EMA UpdateByOperation.") from e
+
+
+def ema_time_decay(ts_col: str, time_scale: Union[int, str], cols: Union[str, List[str]],
+                   op_control: OperationControl = None) -> UpdateByOperation:
+    """Creates an EMA(exponential moving average) UpdateByOperation for the supplied column names, using time as the
+    decay unit.
+
+    The formula used is
+        a = e^(-dt / time_scale_nanos)
+        ema_next = a * ema_last + (1 - a) * value
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+
+        time_scale (Union[int, str]): the decay rate, can be expressed as an integer in nanoseconds or a time
+            interval string, e.g. "00:00:00.001"
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col = col"
+        op_control (OperationControl): defines how special cases should behave,  when None, the default OperationControl
+            settings as specified in :meth:`~OperationControl.__init__` will be used
+
+    Returns:
+        an UpdateByOperation
+
+    Raises:
+        DHError
+     """
+    try:
+        if isinstance(time_scale, str):
+            time_scale = _JDateTimeUtils.expressionToNanos(time_scale)
+        cols = to_sequence(cols)
+        if op_control is None:
+            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.Ema(ts_col, time_scale, *cols))
+        else:
+            return UpdateByOperation(
+                j_updateby_op=_JUpdateByOperation.Ema(op_control.j_op_control, ts_col, time_scale, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a time-decay EMA UpdateByOperation.") from e
+
+
+def cum_sum(cols: Union[str, List[str]]) -> UpdateByOperation:
+    """Creates a cumulative sum UpdateByOperation for the supplied column names.
```

Comment:
Here we removed "CumSum".  We did not do that in the other doc strings.  We need to be consistent.